nodejs/week3/Mealsharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,202 @@
+                              } // end of meal with reservation
+                              else if (availableReservations == "false") {
+                                        // if reservation does not exists then all meals
+                                        const allMeal = await knex(
+                                                  "meal"
+                                        ).select("*");
+                                        res.send(allMeal);
+                              }
+                    } // end  of available reservations
+                    //search by title
+                    if (title) {
+                              const mealTitle = await knex
+                                        .from("meal")
+                                        .select("*")
+                                        .where("title", "like", `%${title}%`);
+                              if (mealTitle.length == 0) {
+                                        res.status(404).send(
+                                                  "Sorry data does not exists any more"
+                                        );
+                              }
+
+                              res.send(mealTitle);
+                    } //end ot title search
+                    // searh  by limit
+
+                    if (limit) {
+                              if (limit) {
+                                        const getLimit = parseInt(limit);
+                                        const mealWithLimit = await knex
+                                                  .select("*")
+                                                  .from("meal")
+                                                  .limit(getLimit);
+                                        res.send(mealWithLimit);
+                              } else {
+                                        res.status(404).send(
+                                                  "Record does not exists"
+                                        );
+                              }
+                    } // end of limit
+                    if (createAfter) {
+                              if (createAfter) {
+                                        const getDate = createAfter;
+                                        const mealWithDate = await knex(
+                                                  "meal"
+                                        ).where("created", "<", `${getDate}`);
+                                        res.send(mealWithDate);
+                              } else {
+                                        res.status(404).send(
+                                                  "Record does not exists"
+                                        );
+                              }
+                    } //end of createdDate
+          } catch (err) {
+                    // end of try block
+                    //Catch block
+                    throw err;
+          }
+});
```

Comment:
All these conditions should be applied to the same query, and not to 4 different queries. They all should be computed together and used to constrain a SELECT query to the database.

Try storing the query knex('meals') in a variable a successively apply `where` statements to it based on the query params, or combine all the query params inside the same `where` call.